src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.cs

```diff
@@ -230,10 +230,10 @@ protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNu
         /// <inheritdoc/>
         protected internal override void SetPinMode(int pinNumber, PinMode mode)
         {
-            int requestResult = -1;
             if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle? pinHandle))
             {
-                requestResult = mode switch
+                Interop.libgpiod.gpiod_line_release(pinHandle);
```

Comment:
why do you even need the interop? Should you just Dispose the handle?

I find this confusing you're releasing native resource and then use handle further...

Reply:
I agree with @krwq here that this indeed looks very fishy. Can you point to the documentation you found where the right thing to do is to call release on the handle and later using it again? The docs I found do point to saying that running this operation will invalidate the handle which then gets used again so I don't think this is the way to go. Also, even if we do have to call the release method, I would never call it directly from here, and instead dispose the safehandle by calling Dispose() and letting the release happen there.

Reply:
I can't find unambiguous documentation on it, unfortunately. But looking at the code [(here)](https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/tree/lib/core.c) it does seem like the right thing to do. `gpiod_line_release` does not free the line handle. Actually, no handles are ever freed until `gpiod_line_close` is called, it seems.

Reply:
If so then that would be a potential bug in our implementation of libgpiod driver because we could have a memory leak by not appropriately freeing the handle on Dispose.

Reply:
No, because it seems the line handle _cannot_ be properly freed. So we could even remove the SafeLineHandle class, as the line handle is cached in the library. Repeatedly calling `gpiod_chip_get_line` always returns the same value as long as the chip is not released. Let's discuss in todays call.